### ISSUE TITLE
Mock.As<T> triggers Moq1300 (Mock.As<T> should be interface) in generic method

### ISSUE THREAD

**Issue by andrewimcclement** [rjmurillo/moq.analyzers#756]
```csharp
interface IFoo;

void SetupMyMock<T>(Mock<IFoo> mock, Action<Mock<T>> setupStuff)
    where T : IFoo
    => setupStuff(mock.As<T>()); // Triggers Moq1300
```

This is admittedly a pretty unusual scenario. However, I feel Moq1300 should not trigger when `T` is a generic type parameter, since there is no way in C# to enforce that `T` is an interface at compile time.

Preferred resolution: The check for Moq1300 either ignores `Mock.As<T>` for `T` a generic type parameter, or raises a separately numbered issue (with potentially different severity).
